glom/tutorial.py

```diff
@@ -215,6 +215,33 @@
 
 Voila, the target can still be parsed and we can elegantly handle changes in our data formats.
 
+Data-Driven Assignment
+======================
+
+Quite often APIs deliver data in dictionaries without constant key values.
+They use parts of the data itself as a key. This we call data-driven assignment.
+
+The following example shows you a way to handle this situation.
+It extracts the moon count from a dictionary that has the planet names as a key.
+
+  >>> from glom import glom, T, merge, Iter, Coalesce
+  >>> target = {
+  ...    "pluto": {"moons": 6, "population": None},
+  ...    "venus": {"population": {"aliens": 5}},
+  ...    "earth": {"moons": 1, "population": {"humans": 7_700_000_000, "aliens": 1}},
+  ... }
+  >>> spec = {
+  ...     "moons": (
+  ...          T.items(),
+  ...          Iter({T[0]: (T[1], Coalesce("moons", default=0))}).all(),
+  ...          merge,
+  ...     )
+  ... }
+  >>> pprint(glom(target, spec))
+  {'moons': {'pluto': 6, 'venus': 0, 'earth': 1}}
+
```

Comment:
I like to follow an example with an illustrative statement. In this case I might say, "Note how we defaulted moons to 0 for venus, which did not have that key set." Or maybe something about using items to access the key with `T[0]` or `Merge()`.